src/containers.js

```diff
@@ -0,0 +1,142 @@
+import {bytesToString, toUint8} from './byte-helpers.js';
+
+export const getId3Offset = function(bytes) {
+  bytes = toUint8(bytes);
+
+  if (bytesToString(bytes.subarray(0, 3)) !== 'ID3') {
+    return 0;
+  }
+  const returnSize = (bytes[6] << 21) |
+                     (bytes[7] << 14) |
+                     (bytes[8] << 7) |
+                     (bytes[9]);
+  const flags = bytes[5];
+  const footerPresent = (flags & 16) >> 4;
+
+  if (footerPresent) {
+    return returnSize + 20;
+  }
+  return returnSize + 10;
+};
+
+export const isLikely = {
+  aac(bytes) {
+    const offset = getId3Offset(bytes);
+
+    return bytes.length >= offset + 2 &&
+      (bytes[offset] & 0xFF) === 0xFF &&
+      (bytes[offset + 1] & 0xE0) === 0xE0 &&
+      (bytes[offset + 1] & 0x16) === 0x10;
+  },
+
+  mp3(bytes) {
+    const offset = getId3Offset(bytes);
+
+    return bytes.length >= offset + 2 &&
+      (bytes[offset] & 0xFF) === 0xFF &&
+      (bytes[offset + 1] & 0xE0) === 0xE0 &&
+      (bytes[offset + 1] & 0x06) === 0x02;
+  },
+
+  webm(bytes) {
+    return bytes.length >= 4 &&
+      (bytes[0] & 0xFF) === 0x1A &&
+      (bytes[1] & 0xFF) === 0x45 &&
+      (bytes[2] & 0xFF) === 0xDF &&
+      (bytes[3] & 0xFF) === 0xA3;
+  },
+  mp4(bytes) {
+    return bytes.length >= 8 &&
+      (/^(f|s)typ$/).test(bytesToString(bytes.subarray(4, 8))) &&
+      // not 3gp data
+      !(/^ftyp3g$/).test(bytesToString(bytes.subarray(4, 10)));
+  },
+
+  '3gp'(bytes) {
+    return bytes.length >= 10 &&
+      (/^ftyp3g$/).test(bytesToString(bytes.subarray(4, 10)));
+  },
+
+  ts(bytes) {
+    return (bytes.length >= 189 && bytes[0] === 0x47 && bytes[188] === 0x47) ||
```

Comment:
Technically, mux.js accounts for misaligned ts streams: https://github.com/videojs/mux.js/blob/master/lib/m2ts/m2ts.js#L69

Should we also try to account for them?

Reply:
Thinking about it some more I think that it shouldn't really be valid or we can't handle it. I think mux.js does that because it actually handles full segments and sometimes we lose which byte we are on somehow (edited) 
4:37
I think the user of isLikely.ts needs to make sure that they are passing in the first 188 bytes of a ts segment for it to work correctly.